benchmarks/persistence/persistence.py

```diff
@@ -0,0 +1,229 @@
+"""Classes for persisting benchmark results."""
+
+import json
+import os
+from typing import Protocol
+
+import boto3
+import boto3.session
+from attr import define, field
+from attrs.validators import instance_of
+from git import Repo
+
+from benchmarks import Benchmark, Result
+
+
+class PersistenceObjectInterface(Protocol):
+    """Interface for constructing the path of a result object."""
+
+    def get_full_file_path(self) -> str:
```

Comment:
Should this be a `str` or a `Path`?

Reply:
Your suggestion on using Path is a really good point thanks for that. I've added it to construct the string depending on the underlying operating system (As far as a know, Windows support POSTIX paths) but still return the string since S3-Buckets does not have a path, if that is fine for you.

Reply:
Did you maybe forget to push here? I cannot find any mention of "Path", or am I missing something?